saleor/graphql/product/bulk_mutations/products.py

```diff
@@ -85,6 +85,9 @@ class Meta:
     def bulk_action(cls, info, queryset):
         delete_categories(queryset.values_list("pk", flat=True), info.context.plugins)
 
+        for instance in queryset:
+            info.context.plugins.category_deleted(instance)
```

Comment:
Keep in mind, that your `instances` here are already deleted, so some part of data could not be accessible here.

Reply:
Also I just quickly look on the tests like `saleor/plugins/webhook/tests/subscription_webhooks/test_create_deliveries_for_subscription.py::test_category_deleted` 
I think that we could have here a fail success. 
You're testing building a logic based on object **that exists in database**. But in real evn, this logic will be executed on object that was already deleted, so some fields could not be accesible.

Reply:
I can confirm. The subscription from test `test_category_deleted` returns `base64(Category:None)` as an ID, when object doesn't exists.

Reply:
I moved `info.context.plugins.category_deleted(instance)` to `delete_categories` function and before I created instances list before `.delete()`.
I updated `test_create_deliveries_for_subscription.py::test_category_deleted` test to use the same flow and I added assert to check if `global_id` is not `None`